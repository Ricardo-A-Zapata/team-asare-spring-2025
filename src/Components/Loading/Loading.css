.loading {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 2rem;
    margin: 2rem auto;
    max-width: 600px;
    min-height: 200px;
    background-color: rgba(0, 0, 0, 0.05);
    border-radius: 8px;
    text-align: center;
  }
  
  .spinner {
    width: 40px;
    height: 40px;
    border: 4px solid rgba(0, 0, 0, 0.1);
    border-radius: 50%;
    border-top-color: var(--accent-color, #4caf50);
    animation: spin 1s linear infinite;
    margin-bottom: 1rem;
  }
  
  @keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }
  