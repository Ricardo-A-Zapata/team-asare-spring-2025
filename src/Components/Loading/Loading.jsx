import React from 'react';
import PropTypes from 'prop-types';

function Loading({ message = "Loading..." }) {
  return (
    <div className="loading">
      <div className="spinner"></div>
      <p>{message}</p>
    </div>
  );
}

Loading.propTypes = {
  message: PropTypes.string
};

export default Loading;
