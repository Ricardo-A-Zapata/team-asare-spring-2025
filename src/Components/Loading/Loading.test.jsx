import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Loading from './Loading';

describe('Loading Component', () => {
  test('renders default loading message', () => {
    render(<Loading />);
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
  });

  test('renders custom loading message', () => {
    render(<Loading message="Fetching data..." />);
    expect(screen.getByText(/Fetching data.../i)).toBeInTheDocument();
  });

  test('renders spinner element', () => {
    const { container } = render(<Loading />);
    expect(container.querySelector('.spinner')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
