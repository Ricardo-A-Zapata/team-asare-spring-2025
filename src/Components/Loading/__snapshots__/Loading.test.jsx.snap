// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="spinner"
    />
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
